File: CS/UnboundColumnWithIcons_MVVM/ViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.Xpf;
using System;
using System.Collections.ObjectModel;
using System.Windows.Media.Imaging;

namespace UnboundColumnWithIcons_MVVM {
    public class MyObject {
        public string Action { get; set; }
        public MyObject(string action) {
            Action = action;
        }
    }

    public class ViewModel : ViewModelBase {
        public ObservableCollection<MyObject> Items { get; set; }

        public ViewModel() {
            Items = new ObservableCollection<MyObject> {
                new MyObject("cut"),
                new MyObject("copy"),
                new MyObject("paste"),
                new MyObject("delete")
            };
        }

        [Command]
        public void UnboundColumnData(UnboundColumnRowArgs args) {
            if(args.FieldName == "IconUnbound" && args.IsGetData) {
                var item = (MyObject)args.Item;
                var resourceName = GetResourceName(item.Action);
                args.Value = GetImage(resourceName);
            }
        }

        BitmapFrame GetImage(string resourceName) {
            var uri = new Uri("pack://application:,,,/Icons/" + resourceName, UriKind.Absolute);
            return BitmapFrame.Create(uri);
        }
        string GetResourceName(string action) {
            switch(action) {
                case "copy":
                    return "copy32x32.png";
                case "cut":
                    return "cut32x32.png";
                case "delete":
                    return "delete32x32.png";
                case "paste":
                    return "paste32x32.png";
                default:
                    return string.Empty;
            }
        }
    }
}

File: VB/UnboundColumnWithIcons_MVVM/ViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.Xpf
Imports System
Imports System.Collections.ObjectModel
Imports System.Windows.Media.Imaging

Namespace UnboundColumnWithIcons_MVVM

    Public Class MyObject

        Public Property Action As String

        Public Sub New(ByVal action As String)
            Me.Action = action
        End Sub
    End Class

    Public Class ViewModel
        Inherits ViewModelBase

        Public Property Items As ObservableCollection(Of MyObject)

        Public Sub New()
            Items = New ObservableCollection(Of MyObject) From {New MyObject("cut"), New MyObject("copy"), New MyObject("paste"), New MyObject("delete")}
        End Sub

        <Command>
        Public Sub UnboundColumnData(ByVal args As UnboundColumnRowArgs)
            If args.FieldName Is "IconUnbound" AndAlso args.IsGetData Then
                Dim item = CType(args.Item, MyObject)
                Dim resourceName = GetResourceName(item.Action)
                args.Value = GetImage(resourceName)
            End If
        End Sub

        Private Function GetImage(ByVal resourceName As String) As BitmapFrame
            Dim uri = New Uri("pack://application:,,,/Icons/" & resourceName, UriKind.Absolute)
            Return BitmapFrame.Create(uri)
        End Function

        Private Function GetResourceName(ByVal action As String) As String
            Select Case action
                Case "copy"
                    Return "copy32x32.png"
                Case "cut"
                    Return "cut32x32.png"
                Case "delete"
                    Return "delete32x32.png"
                Case "paste"
                    Return "paste32x32.png"
                Case Else
                    Return String.Empty
            End Select
        End Function
    End Class
End Namespace
